### PR TITLE
ci: use vite preview

### PR DESCRIPTION
**Pull request by himself65**
image

### CODE REVIEW

**Comment by doodlewind** [comment]
How does this work? Does this mean for each time entering the playground page, the bundle built by `vite preview` will be pre-packaged?

**Comment by himself65** [comment]
> How does this work? Does this mean for each time entering the playground page, the bundle built by `vite preview` will be pre-packaged?

yep, playground would output to a single zip, so that you can download to our local

**Comment by himself65** [comment]
> > How does this work? Does this mean for each time entering the playground page, the bundle built by `vite preview` will be pre-packaged?
> 
> yep, playground would output to a single zip, so that you can download to our local

also this would speed up the e2e test because vite build output is production version

**Comment by doodlewind** [comment]
Got it (https://stackoverflow.com/questions/71703933/what-is-the-difference-between-vite-and-vite-preview), this makes sense.

File: .github/workflows/cancel.yml [hunk 1]
```diff
@@ -13,5 +13,5 @@ jobs:
     steps:
       - uses: styfle/cancel-workflow-action@0.11.0
         with:
-          workflow_id: 46388772, 34612818
```
Comment (himself65) [inline .github/workflows/cancel.yml:16]: see https://api.github.com/repos/toeverything/blocksuite/actions/workflows

File: .github/workflows/build-and-test.yaml [hunk 2]
```diff
@@ -21,10 +21,51 @@ on:
       - 'packages/react/examples/**'
 
 jobs:
-  main-test:
```
Comment (himself65) [inline .github/workflows/build-and-test.yaml:24]: move build and test into a single file

**Review by doodlewind — APPROVED** [review]

**Pull request merged**